src/language/nodes.py

```diff
@@ -276,26 +286,94 @@ def get_setter_method(self, class_name):
                        /**
                         * \\brief Setter for member variable \\ref {class_name}.{self.varname}
                         */
-                       void {setter_method}({setter_type} {self.varname}) {{
-                           this->{self.varname} = {self.varname};
-                       }}
+                       void {setter_method}({setter_type} {self.varname});
                     """
         else:
             return f"""
                        /**
                         * \\brief Setter for member variable \\ref {class_name}.{self.varname} (rvalue reference)
                         */
-                       void {setter_method}({setter_type}&& {self.varname}) {{
-                           this->{self.varname} = {self.varname};
-                       }}
-                       
+                       void {setter_method}({setter_type}&& {self.varname});
+
                        /**
                         * \\brief Setter for member variable \\ref {class_name}.{self.varname}
                         */
-                       void {setter_method}(const {setter_type}& {self.varname}) {{
+                       void {setter_method}(const {setter_type}& {self.varname});
+                    """
+
+
+
+    def get_setter_method_definition(self, class_name):
+        setter_method = "set_" + to_snake_case(self.varname)
+        setter_type = self.member_typename
+        reference = "" if self.is_base_type_node else "&&"
+
+
+        if self.is_base_type_node:
+            return f"""
+                       void {class_name}::{setter_method}({setter_type} {self.varname}) {{
+                           // why don't we use a coding convention instead of this workaround for
+                           // variable shadowing?
```

Comment:
do you mean to avoid this-> ? Yes we could. It was just simple to use same name in the template without prefix/suffix.

Reply:
ok, for now I leave it. The commit/pr is about something else